### ISSUE TITLE
When the form has initial data and the schema is updated, the form throws an error

### ISSUE THREAD

**Issue by bhch** [bhch/django-jsonform#131]
Steps to reproduce:

Step 1. Original schema:

```json
{
  "type": "object",
  "properties": {
    "data": {
      "type": "object",
      "title": "Example",
      "oneOf": [
        {
          "title": "Table",
          "properties": {
            "type": {
              "type": "string",
              "default": "table",
              "widget": "hidden"
            },
            "columns": {
              "title": "Columns",
              "type": "array",
              "items": {
                "type": "object",
                "properties": {
                  "title": {
                    "type": "string",
                    "title": "Header"
                  },
                  "type": {
                    "type": "string",
                    "title": "Type",
                    "choices": [
                      "string",
                      "number",
                      "integer",
                      "boolean"
                    ],
                    "default": "string",
                    "required": true
                  },
                  "default": {
                    "type": "array",
                    "title": "Prepopulate rows (EN)",
                    "items": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          }
        }
      ]
    }
  }
}
```

Step 2: Add some data and save the object.

Step 3: Update the schema to this:

```json
{
  "type": "object",
  "properties": {
    "data": {
      "type": "object",
      "title": "Example",
      "oneOf": [
        {
          "title": "Table",
          "properties": {
            "type": {
              "type": "string",
              "default": "table",
              "widget": "hidden"
            },
            "columns": {
              "title": "Columns",
              "type": "array",
              "items": {
                "type": "object",
                "properties": {
                  "title": {
                    "type": "string",
                    "title": "Header"
                  },
                  "type": {
                    "type": "string",
                    "title": "Type",
                    "choices": [
                      "string",
                      "number",
                      "integer",
                      "boolean"
                    ],
                    "default": "string",
                    "required": true
                  },
                  "default": {
                    "type": "array",
                    "title": "Prepopulate rows (EN)",
                    "items": {
                      "type": "string"
                    }
                  },
                  "default_de": {
                    "type": "array",
                    "title": "Prepopulate rows (DE)",
                    "items": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          }
        }
      ]
    }
  }
}
```

Step 4: Refresh page

The admin widget doesn't render with the following message:
<img width="570" alt="Screenshot 2023-11-28 at 10 02 20 AM" src="https://github.com/bhch/django-jsonform/assets/141132507/db6f21b9-f41f-4923-aee5-bca0304f8d11">

I was using 2.17.0 earlier but upgraded to 2.21.0 but the problem still persists.

_Originally posted by @sgocg in https://github.com/bhch/django-jsonform/issues/101#issuecomment-1829378697_

**Comment by bhch** [comment]
@sgocg This has been fixed in [v2.21.1](https://github.com/bhch/django-jsonform/releases/tag/v2.21.1). Please upgrade and test it out.

Thank you for reporting the issue.

**Comment by sgocg** [comment]
Amazing, thanks for the fix! I tested it locally and it works quite well.
